iotdb/src/main/java/org/apache/iotdb/db/query/reader/sequence/SealedTsFilesReaderByTimestamp.java

```diff
@@ -0,0 +1,137 @@
+import org.apache.iotdb.db.query.control.FileReaderManager;
+import org.apache.iotdb.db.query.reader.merge.EngineReaderByTimeStamp;
+import org.apache.iotdb.db.utils.QueryUtils;
+import org.apache.iotdb.tsfile.file.metadata.ChunkMetaData;
+import org.apache.iotdb.tsfile.read.TsFileSequenceReader;
+import org.apache.iotdb.tsfile.read.common.Path;
+import org.apache.iotdb.tsfile.read.controller.ChunkLoader;
+import org.apache.iotdb.tsfile.read.controller.ChunkLoaderImpl;
+import org.apache.iotdb.tsfile.read.controller.MetadataQuerierByFileImpl;
+import org.apache.iotdb.tsfile.read.reader.series.SeriesReaderByTimestamp;
+
+public class SealedTsFilesReaderByTimestamp implements EngineReaderByTimeStamp {
+
+  private Path seriesPath;
+  private List<IntervalFileNode> sealedTsFiles;
+  private int nextIntervalFileIndex;
+  private SeriesReaderByTimestamp seriesReader;
+  private QueryContext context;
+
+  /**
+   * init with seriesPath and sealedTsFiles.
+   */
+  public SealedTsFilesReaderByTimestamp(Path seriesPath, List<IntervalFileNode> sealedTsFiles,
+      QueryContext context) {
+    this.seriesPath = seriesPath;
+    this.sealedTsFiles = sealedTsFiles;
+    this.nextIntervalFileIndex = 0;
+    this.seriesReader = null;
+    this.context = context;
+  }
+
+  @Override
+  public Object getValueInTimestamp(long timestamp) throws IOException {
+    Object value = null;
+    if (seriesReader != null) {
+      value = seriesReader.getValueInTimestamp(timestamp);
+      if (value != null || seriesReader.hasNext()) {
+        return value;
+      }
+    }
+    constructReader(timestamp);
+    if (seriesReader != null) {
+      value = seriesReader.getValueInTimestamp(timestamp);
+      if (value != null || seriesReader.hasNext()) {
+        return value;
+      }
+    }
+
+    return value;
+  }
+
+  @Override
+  public boolean hasNext() throws IOException {
+    if (seriesReader != null && seriesReader.hasNext()) {
+      return true;
+    }
+    while (nextIntervalFileIndex < sealedTsFiles.size()) {
+      initSingleTsFileReader(sealedTsFiles.get(nextIntervalFileIndex), context);
+      nextIntervalFileIndex++;
+      if(seriesReader.hasNext()){
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @Override
+  public void close() throws IOException {
+    // file streams are managed uniformly.
+  }
+
+  // construct reader from the file that might overlap this timestamp
+  private void constructReader(long timestamp) throws IOException {
+    while (nextIntervalFileIndex < sealedTsFiles.size()) {
+      if (singleTsFileSatisfied(sealedTsFiles.get(nextIntervalFileIndex), timestamp)) {
+        initSingleTsFileReader(sealedTsFiles.get(nextIntervalFileIndex), context);
+      }
+      nextIntervalFileIndex++;
+    }
+  }
+
+  /**
+   * Judge whether the file should be skipped.
+   */
+  private boolean singleTsFileSatisfied(IntervalFileNode fileNode, long timestamp) {
+    long endTime = fileNode.getEndTime(seriesPath.getDevice());
```

Comment:
`startTime` also needed?

Reply:
SealedTsFilesReaderByTimestamp acquires data from small to large according to the timestamp, so as long as the maximum timestamp is greater than or equal to the timestamp, the file should not be skipped.